### PR TITLE
Add banner to account page

### PR DESCRIPTION
**Pull request by DilwoarH**
Add the user research banner to the account page using the existing GOV.uk pattern. Allow for links to 'sign up to mailing list' page

As a  user I should be presented with thte banner to sign up to the user research mailing list when I log in if: I have not already closed it and I have not already signed up for this.
This is the first point of contact for getting users to sign up to the mailing list.

The close link should set a cookie value of 'seen_this_banner_thing' to true so we don't show it again
The sign up link should link to the new sign up page.

Ticket: https://trello.com/c/OJ9lFk1M/129-add-banner-to-account-page

Documentation:
screen shot 2018-03-06 at 16 51 03

### CODE REVIEW

File: toolkit/scss/_user-research-consent-banner.scss [hunk 1]
```diff
@@ -0,0 +1,29 @@
+.user-research-banner {
+  background-color: #005da5;
```
Comment (idavidmcdonald) [inline toolkit/scss/_user-research-consent-banner.scss]: Can all our colours be SASS variables please

File: toolkit/javascripts/user-research-consent-banner.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+;(function (global) {
+  "use strict";
+
+  var UserResearchBanner = function( _userResearchBanner ) {
+    this.$userResearchBanner = _userResearchBanner;
+    this.$userResearchBannerCloseBtn = this.$userResearchBanner.find('.user-research-banner-close-btn');
+    this.userResearchCookie = {
+      name: 'seen_user_research_message',
```
Comment (idavidmcdonald) [inline toolkit/javascripts/user-research-consent-banner.js]: Nice work with consistency with the gov.uk cookies cookie

File: toolkit/templates/user-research-consent-banner.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<div class="user-research-banner">
```
Comment (idavidmcdonald) [inline toolkit/templates/user-research-consent-banner.html]: We could instead use `id` as opposed to `class` as we should only be having one of these. The GOV.UK cookie banner seems to use `id="global-cookie-message`. I don't feel too strongly on this though. Happy to go with what you feel is best.

File: toolkit/templates/user-research-consent-banner.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  <div class="wrapper">
+    <p>
+      <strong class="banner-heading">Help us improve the Digital Marketplace</strong>
+      <a href="#" class="user-research-banner-close-btn">Close</a>
+    </p>
+    <p>
+      <a href="#">Sign up to be a potential user research participants</a>
```
Comment (idavidmcdonald) [inline toolkit/templates/user-research-consent-banner.html]: Any idea where this link is going?

File: toolkit/templates/user-research-consent-banner.html (hunk 3 above)
Comment (DilwoarH) [reply toolkit/templates/user-research-consent-banner.html]: I don't feel too strongly but I tend to use classes where ever possible and only IDs when actually required. Conflicting IDs can sometimes break older browsers

**Comment by DilwoarH** [comment]
@idavidmcdonald - I have addressed the comments

File: toolkit/templates/user-research-consent-banner.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<div class="user-research-banner">
+  <div class="wrapper">
+    <p>
+      <strong class="banner-heading">Help us improve the Digital Marketplace</strong>
+      <a href="#" class="user-research-banner-close-btn">Close</a>
+    </p>
+    <p>
+      <a href="{{ url_for('external.user_research_consent') }}">Sign up to be a potential user research participants</a>
```
Comment (idavidmcdonald) [inline toolkit/templates/user-research-consent-banner.html]: Is the `s` on the end of the sentence correct?

Comment (DilwoarH) [reply toolkit/templates/user-research-consent-banner.html]: Good catch

File: toolkit/javascripts/user-research-consent-banner.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+  UserResearchBanner.prototype.showBanner = function () {
+    this.$userResearchBanner.show();
+  };
+
+  UserResearchBanner.prototype.hideBanner = function () {
+    this.$userResearchBanner.hide();
+  };
+
+  UserResearchBanner.prototype.getCookie = function () {
```
Comment (benvand) [inline toolkit/javascripts/user-research-consent-banner.js]: Will this cause the banner to show if the user isn't logged in?

Comment (DilwoarH) [reply toolkit/javascripts/user-research-consent-banner.js]: The template should not be inserted if the user is anonymous. Not sure if there is some JS helper that allows us to see if the user is logged in or not

**Pull request merged**